clusterloader2/pkg/measurement/common/wait_for_controlled_pods.go

```diff
@@ -331,9 +331,10 @@ func (w *waitForControlledPodsRunningMeasurement) deleteObjectLocked(obj runtime
 	if err != nil {
 		return fmt.Errorf("meta key creation error: %v", err)
 	}
-	if checker, exists := w.checkerMap[key]; exists {
-		checker.terminate(false)
-		delete(w.checkerMap, key)
+	cm := (map[string]checker.Checker)(w.checkerMap)
+	if checker, exists := cm[key]; exists {
```

Comment:
DeleteAndStopIf(key) bool
should be added to the interface and used here.

Reply:
Done. Although I didn't add the return type.